packages/ckeditor5-ui/src/colorpicker/colorpickerview.ts

```diff
@@ -0,0 +1,342 @@
+		super( locale );
+
+		this.set( 'color', '' );
+
+		this.set( '_hexColor', '' );
+
+		this._format = config.format || 'hsl';
+
+		this.hexInputRow = this._createInputRow();
+
+		const children = this.createCollection();
+		children.add( this.hexInputRow );
+
+		this.setTemplate( {
+			tag: 'div',
+			attributes: {
+				class: [ 'ck', 'ck-color-picker' ],
+				tabindex: -1
+			},
+			children
+		} );
+
+		this._debounceColorPickerEvent = debounce( ( color: string ) => {
+			this.set( 'color', color );
+		}, waitingTime, {
+			leading: true
+		} );
+
+		// Sets color in the picker if color was updated.
+		this.on( 'set:color', ( evt, propertyName, newValue ) => {
+			// The color needs always to be kept in the output format.
+			evt.return = convertColor( newValue, this._format );
+		} );
+
+		this.on( 'change:color', () => {
+			this._hexColor = convertColorToCommonHexFormat( this.color );
+		} );
+
+		this.on( 'change:_hexColor', () => {
+			this.picker.setAttribute( 'color', this._hexColor );
+
+			// There has to be two way binding between properties.
+			// Extra precaution has to be taken to trigger change back only when the color really changes.
+			if ( convertColorToCommonHexFormat( this.color ) != convertColorToCommonHexFormat( this._hexColor ) ) {
+				this.color = this._hexColor;
+			}
+		} );
+	}
+
+	/**
+	 * Renders color picker in the view.
+	 */
+	public override render(): void {
+		super.render();
+
+		this.picker = global.document.createElement( 'hex-color-picker' );
+		this.picker.setAttribute( 'class', 'hex-color-picker' );
+		this.picker.setAttribute( 'tabindex', '-1' );
+
+		this._createSlidersView();
+
+		if ( this.element ) {
+			this.element.insertBefore( this.picker, this.hexInputRow.element );
+
+			// Create custom stylesheet with a look of focused pointer in color picker and append it into the color picker shadowDom
+			const styleSheetForFocusedColorPicker = document.createElement( 'style' );
+
+			styleSheetForFocusedColorPicker.textContent = '[role="slider"]:focus [part$="pointer"] {' +
+					'border: 1px solid #fff;' +
+					'outline: 1px solid var(--ck-color-focus-border);' +
+					'box-shadow: 0 0 0 2px #fff;' +
+				'}';
+			this.picker.shadowRoot!.appendChild( styleSheetForFocusedColorPicker );
+		}
+
+		this.picker.addEventListener( 'color-changed', event => {
+			const customEvent = event as CustomEvent;
+			const color = customEvent.detail.value;
+			this._debounceColorPickerEvent( color );
+		} );
+	}
+
+	/**
+	 * Focuses the first pointer in color picker.
+	 *
+	 */
+	public focus(): void {
+		const firstSlider = this.slidersView.first!;
+
+		firstSlider.focus();
+	}
+
+	/**
+	 * Creates collection of sliders in color picker.
+	 *
+	 * @private
+	 */
+	private _createSlidersView(): void {
+		const colorPickersChildren = [ ...this.picker.shadowRoot!.children ] as Array<HTMLElement>;
+		const sliders = colorPickersChildren.filter( item => item.role === 'slider' );
```

Comment:
This filtering doesn't work on Firefox. It returns an empty array. It causes https://github.com/cksource/ckeditor5-internal/issues/3243

Reply:
It seems enough to change `.role` to `.getAttribute( 'role' )` to make it work. It works in Chrome and Firefox.

Reply:
Thanks! Indeed it did the trick.